cluster-controller/src/main/java/io/strimzi/controller/cluster/model/KafkaCluster.java

```diff
@@ -290,6 +298,7 @@ public ConfigMap generateMetricsConfigMap() {
         List<EnvVar> varList = new ArrayList<>();
         varList.add(buildEnvVar(KEY_KAFKA_ZOOKEEPER_CONNECT, zookeeperConnect));
         varList.add(buildEnvVar(KEY_KAFKA_METRICS_ENABLED, String.valueOf(isMetricsEnabled)));
+        kafkaHeapOptions(varList, 0.5, 5L * 1024L * 1024L * 1024L);
```

Comment:
@tombentley maybe I miss some discussion but from where these values come from ?

Reply:
Those values don't particularly come from anywhere, except to say that Confluent and Kafka docs suggest there's not much point in running Kafka brokers with heap > 5bg, and for kafka we want to "leave some space for page cache", so we default to using 50% of the limit for the jvm memory upto a limit of 5gb heap. 

The values for ZK and Connect are more "finger in the air".

In reality what we're going to need next is better tools/tests/mgmt to be able to have visibility of jvm memory usage and gc for all the containers. But not in this PR.